GooglePlayInstant/Editor/PlayInstantQuickDeployInputManager.cs

```diff
@@ -23,97 +21,125 @@ namespace GooglePlayInstant.Editor
+        public static string RemoteBucketName
         {
-            if (string.Equals(field, null))
+            get
             {
-                RetrieveTextFieldInputs();
+                if (_remoteBucketName != null)
+                {
+                    return _remoteBucketName;
+                }
+
+                UpdateFieldFromPrefs(ref _remoteBucketName, "remoteBucketName");
+                return _remoteBucketName;
             }
+            set
+            {
+                if (string.Equals(_remoteBucketName, value))
+                {
+                    return;
+                }
 
-            return field;
+                _remoteBucketName = value;
+                UpdatePrefs("remoteBucketName", value);
+            }
         }
 
-        // Do nothing if value has not changed, otherwise update the value of the static field and write to file.
-        private static void updateNewValue(ref string field, string value)
+        public static string RemoteObjectName
         {
-            if (string.Equals(field, value))
+            get
             {
-                return;
+                if (_remoteObjectName != null)
+                {
+                    return _remoteObjectName;
+                }
+
+                UpdateFieldFromPrefs(ref _remoteObjectName, "remoteObjectName");
+                return _remoteObjectName;
             }
+            set
+            {
+                if (string.Equals(_remoteObjectName, value))
+                {
+                    return;
+                }
 
-            field = value;
-            StoreTextFieldInputs();
+                _remoteObjectName = value;
+                UpdatePrefs("remoteObjectName", value);
+            }
         }
 
-        private static void StoreTextFieldInputs()
+        // Set DataStoragePrefix + key to value in EditorPrefs if value is not empty or null. Otherwise remove key.
+        private static void UpdatePrefs(string key, string value)
         {
-            var fieldsData = new TextFieldInputFileContent
+            if (string.IsNullOrEmpty(value))
             {
-                oauth2CredentialsPath = _oauth2CredentialsPath ?? "path_to_google_cloud_project_oauth2_credentials",
-                localAssetBundlePath = _localAssetBundlePath ?? "local_path_to_assetbundle",
-                remoteBucketName = _remoteBucketName ?? "remote_bucket_name",
-                remoteObjectName = _remoteObjectName ?? "remote_object_name",
-            };
-            File.WriteAllText(dataStorageFilePath, JsonUtility.ToJson(fieldsData));
-        }
-
-        private static void RetrieveTextFieldInputs()
-        {
-            if (!File.Exists(dataStorageFilePath))
+                EditorPrefs.DeleteKey(DataStoragePrefix + key);
+            }
+            else
             {
-                StoreTextFieldInputs();
+                EditorPrefs.SetString(DataStoragePrefix + key, value);
```

Comment:
Should these values be shared between projects? Unless you include another unique ID in the key (e.g. package name) all projects will share this.

P.S. You might be able to find a better way to do per-project configuration storage. EditorPrefs was what I found but Unity may offer better options.

Reply:
Resolved. Abandoned this code since the config pull request addresses this